### ISSUE TITLE
net::ERR_CONTENT_LENGTH_MISMATCH

### ISSUE THREAD

**Issue by pjmcdermott** [cobrateam/django-htmlmin#117]
python 3.5.2, django 1.11.4

I am seeing the above error when loading a page in Chromium browser, and also in Opera. Firefox does not seem to be bothered about it.

On checking the header data in Firefox there does seem to be a discrepancy between the Content-Length header and the the length of the response content. The HTML is minified OK.

Content-Length is first set by Django in `django.middleware.common.CommonMiddleware`, which I have in my `MIDDLEWARE` list.

```
# Add the Content-Length header to non-streaming responses if not
# already set.
if not response.streaming and not response.has_header('Content-Length'):
    response['Content-Length'] = str(len(response.content))
```

Interestingly, there does not seem to be any attempt to correct the Content-Length header in `htmlmin`: `grep -Ri Content-Length lib/python3.5/site-packages/htmlmin/` draws a blank.

Changing the end of  `htmlmin.middleware.HtmlMinifyMiddleware.process_request` to:

```
        if minify and self.can_minify_response(request, response):
            response.content = html_minify(response.content,
                                           ignore_comments=not keep_comments,
                                           parser=parser)
            response['Content-Length'] = str(len(response.content))
        return response
```
appears to solve the problem.

I am surprised more people have not experienced this issue. Is using `django.middleware.common.CommonMiddleware` not actually so common?

**Comment by pjmcdermott** [comment]
[django/docs/ref/middleware.txt](https://github.com/django/django/blob/9588718cd404d73b2be09241b424b539d8d8c66e/docs/ref/middleware.txt) explains that the Content-Length header only started to be set from Django 1.11, which explains why this issue was not noticed previously.

**Comment by mkoistinen** [comment]
I'm seeing it for sure and your proposed fix works great. Thanks for the analysis and fix!

**Comment by andrewsmedina** [comment]
The support for django 1.11 (that fix this issue) was released in the 0.11 version.
